pkg/v1/internal/gzip/zip.go

```diff
@@ -82,15 +83,17 @@ func UnzipReadCloser(r io.ReadCloser) (io.ReadCloser, error) {
 	}, nil
 }
 
-// Is detects whether the input stream is compressed.
-func Is(r io.Reader) (bool, error) {
-	magicHeader := make([]byte, 2)
-	n, err := r.Read(magicHeader)
-	if n == 0 && err == io.EOF {
-		return false, nil
-	}
+// Is detects whether the input stream is gzip compressed. A small number
+// of bytes are buffered to Peek at the gzip header so that the returned
+// io.Reader can be used as a replacement for the consumed input io.Reader.
+func Is(r io.Reader) (bool, io.Reader, error) {
+	buf := bufio.NewReader(r)
+	header, err := buf.Peek(2)
 	if err != nil {
-		return false, err
+		if err == io.EOF {
+			return false, buf, nil
+		}
```

Comment:
Why swallow the EOF?

Reply:
tl;dr the answer to "is this gzipped?" is "no" if we encounter an `io.EOF` by reading the first two bytes.